python/tk_multi_loader/dialog.py

```diff
@@ -363,6 +379,46 @@ def _on_history_double_clicked(self, model_index):
         if default_action:
             default_action.trigger()
 
+    def _on_thumbnail_mode_clicked(self):
+        """
+        Executed when someone clicks the thumbnail mode button
+        """
+        self._set_main_view_mode(self.MAIN_VIEW_THUMB)
+        
+    def _on_list_mode_clicked(self):
+        """
+        Executed when someone clicks the list mode button
+        """
+        self._set_main_view_mode(self.MAIN_VIEW_LIST)
+
+    def _set_main_view_mode(self, mode):
+        """
+        Sets up the view mode for the main view.
+        
+        :param mode: either MAIN_VIEW_LIST or MAIN_VIEW_THUMB
+        """
+        if mode == self.MAIN_VIEW_LIST:
+            self.ui.list_mode.setIcon(QtGui.QIcon(QtGui.QPixmap(":/res/mode_switch_card_active.png")))
+            self.ui.list_mode.setChecked(True)
+            self.ui.thumbnail_mode.setIcon(QtGui.QIcon(QtGui.QPixmap(":/res/mode_switch_thumb.png")))
+            self.ui.thumbnail_mode.setChecked(False)
+            self.ui.publish_view.setViewMode(QtGui.QListView.ListMode)
+            self.ui.publish_view.setItemDelegate(self._publish_list_delegate)            
+            
+        elif mode == self.MAIN_VIEW_THUMB:
+            self.ui.list_mode.setIcon(QtGui.QIcon(QtGui.QPixmap(":/res/mode_switch_card.png")))
+            self.ui.list_mode.setChecked(False)
+            self.ui.thumbnail_mode.setIcon(QtGui.QIcon(QtGui.QPixmap(":/res/mode_switch_thumb_active.png")))
+            self.ui.thumbnail_mode.setChecked(True)
+            self.ui.publish_view.setViewMode(QtGui.QListView.IconMode)
+            self.ui.publish_view.setItemDelegate(self._publish_thumb_delegate)
+            
+        else:
+            raise TankError("Undefined view mode!") 
+
+        self.ui.publish_view.selectionModel().clear()
```

Comment:
It's a delegate weirdness thing. i'll add a comment and will add it to the v3 doc for the libraries :)